hack/lib/util.sh

```diff
@@ -729,13 +729,13 @@ function kube::util::ensure-gnu-sed {
 function kube::util::ensure-gnu-date {
   # NOTE: the echo below is a workaround to ensure date is executed before the grep.
   # see: https://github.com/kubernetes/kubernetes/issues/87251
-  date_help="$(LANG=C date --help 2>&1 || true)"
-  if echo "${date_help}" | grep -q "GNU\|BusyBox"; then
+  date_version="$(LANG=C date --version 2>&1 || true)"
+  if echo "${date_version}" | grep -q "GNU\|BusyBox\|uutils"; then
```

Comment:
Continuing to call it `ensure-gnu-date` but then accepting a non-gnu date seems like it will introduce new bugs.

Reply:
I see that uutils claims to aim for identical output:
https://github.com/uutils/coreutils#goals

Reply:
Yes, sorry, I meant that since it intends to be compatible it's reasonable to rename this / update the message and permit uutils because it intends to be ~"API" compatible, version/license info aside.